src/Api/UserApi.php

```diff
@@ -25,6 +55,94 @@ public function setPandaAuth($value)
    */
   public function userPost(Register $register, string $acceptLanguage = null, &$responseCode, array &$responseHeaders)
   {
-    // TODO: Implement userPost() method.
+    $validation_schema = $this->validate($register);
+
+    if ($validation_schema->getEmail() || $validation_schema->getUsername() || $validation_schema->getPassword())
+    {
+      $responseCode = 422; // 422 => Unprocessable entity
+      return $this->validate($register);
+    }
+    if ($register->isDryRun())
+    {
+      $responseCode = 204; // 204 => Dry-run successful, no validation error
+    }
+    else
+    {
+      // Validation successful, no dry-run requested => we can actually register the user
+      $user = $this->user_manager->createUser();
+      $user->setUsername($register->getUsername());
+      $user->setEmail($register->getEmail());
+      $user->setPlainPassword($register->getPassword());
+      $user->setEnabled(true);
+      $user->setUploadToken($this->token_generator->generateToken());
+      $this->user_manager->updateUser($user);
+      $responseCode = 201; // 201 => User successfully registered
+    }
+  }
+
+  /**
+   * Validates the Register object passed by the request. No automatic validation provided by the OpenApi
+   * will be used cause non standard validations (e.g. validation if a username doesn't exist already) must be
+   * used here.
+   *
+   * @param Register $register The Register object to be validated
+   *
+   * @return ValidationSchema The ValidationSchema containing possible validation errors
+   */
+  public function validate(Register $register)
+  {
+    $validation_schema = new ValidationSchema();
+
+    // E-Mail
+    if (!strlen($register->getEmail()))
+    {
+      $validation_schema->setEmail('EMail missing');
```

Comment:
Since this API hast the `Accept-Language`-Header please add the strings as a translation to the appropriate file and translate the error message to the given locale in `Accept-Language`.

Reply:
Fallback should be `en` 😄